RecentActivity/src/org/sleuthkit/autopsy/recentactivity/Extract.java

```diff
@@ -216,4 +220,206 @@ public boolean foundData() {
     protected void setFoundData(boolean foundData){
         dataFound = foundData;
     }
+    
+    /**
+     * Returns the current case instance
+     * @return Current case instance
+     */
+    protected Case getCurrentCase(){
+        return this.currentCase;
+    }
+    
+    /**
+     * Creates a list of attributes for a history artifact.
+     *
+     * @param url 
+     * @param accessTime Time url was accessed
+     * @param referrer referred url
+     * @param title title of the page
+     * @param programName module name
+     * @param domain domain of the url
+     * @param user user that accessed url
+     * @return List of BlackboardAttributes for giving attributes
+     * @throws TskCoreException
+     */
+    protected Collection<BlackboardAttribute> createHistoryAttribute(String url, Long accessTime,
+            String referrer, String title, String programName, String domain, String user) throws TskCoreException {
+
+        Collection<BlackboardAttribute> bbattributes = new ArrayList<>();
+        bbattributes.add(new BlackboardAttribute(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_URL,
+                RecentActivityExtracterModuleFactory.getModuleName(),
+                (url != null) ? url : ""));
```

Comment:
// NON-NLS

By putting this tag in once as developers, as tedious as it is, we save having a translator have to look at these lines and ignore them every time the file is parsed for translation. AT least I don't think that the parsers ignore empty strings...